docs/source/serving/high-availability.rst

```diff
@@ -0,0 +1,177 @@
+.. _skyserve-high-availability-controller:
+
+=================================
+High Availability SkyServe Controller
+=================================
+
+Overview
+--------
+By default, the SkyServe controller runs as a single instance (either a VM or a Kubernetes Pod). If this instance fails due to node issues, pod crashes, or other unexpected events, the service endpoint becomes unavailable until the controller is manually recovered or relaunched.
+
+To enhance resilience and ensure service continuity, SkyServe offers a High Availability (HA) mode for its controller. When enabled, the controller leverages Kubernetes capabilities to automatically recover from failures.
+
+Benefits of HA Controller:
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+* **Automatic Recovery:** The controller and load balancer can automatically restart after crashes or node failures.
+* **Service Continuity:** Minimizes downtime for your served applications.
+* **Managed by Kubernetes:** Relies on robust, industry-standard Kubernetes features for state management and process supervision.
+
+Prerequisites
+-------------
+* **Kubernetes Cluster:** High Availability mode is **currently only supported when using Kubernetes** as the cloud provider for the SkyServe controller. You must have a Kubernetes cluster configured for SkyPilot. See :ref:`Kubernetes Setup <kubernetes-setup>` for details.
+* **Persistent Kubernetes:** The underlying Kubernetes cluster (control plane and nodes) must be running persistently. If using a local Kubernetes deployment (e.g., Minikube, Kind via `sky local up`), the machine hosting the cluster must remain online.
+
+.. note::
+    Currently, HA mode is only supported for Kubernetes. Support for other clouds (e.g., AWS, GCP, Azure VMs) is under development.
+
+How to Enable HA Mode
+---------------------
+To enable High Availability for the SkyServe controller, set the ``high_availability`` flag to ``true`` within the ``serve.controller`` section of your SkyPilot configuration file (``~/.sky/config.yaml``):
+
+.. code-block:: yaml
+    :emphasize-lines: 8
+
+    serve:
+      # NOTE: these settings only take effect for a *new* SkyServe controller,
+      # not if you have an existing one (even if stopped).
+      controller:
+        # --- Controller Resources ---
+        # Optional: Specify resources like cloud, region, cpus, disk_size
+        resources:
+          cloud: kubernetes  # HA mode requires Kubernetes
+          # region: <your-k8s-region> # Optional
+          cpus: 2+           # Optional, example value
+          # disk_size: 100     # Optional, example value (affects PVC size in HA mode)
+
+        # --- Enable High Availability ---
+        high_availability: true
```

Comment:
Can we add this all to `reference/config.rst` as well?

Reply:
SkyServe config are not added to `config.rst`. All stored seperately in https://docs.skypilot.co/en/latest/serving/sky-serve.html#customizing-skyserve-controller-resources.

Reply:
We have a similar detailed breakdown for managed jobs https://docs.skypilot.co/en/latest/examples/managed-jobs.html#customizing-jobs-controller-resources but I think we should still include all the possible configuration in `config.rst` for completeness. `config.rst` can link back here for details.